azkaban-common/src/main/java/azkaban/scheduler/Schedule.java

```diff
@@ -224,6 +273,29 @@ private DateTime getNextRuntime(long scheduleTime, DateTimeZone timezone,
     return date;
   }
 
+  private DateTime getNextCronRuntime(long scheduleTime, DateTimeZone timezone,
```

Comment:
Could you add comment to explain the logic here? 

My understanding is that you want to avoid returning a datetime that is in the past. Right? What does the scheduleTime represent? What if you pass the later of ( current time and scheduleTime ) to getNextValidTimeAfter? 

Since we don't support schedule a time after a future date, we can simplify it further by passing just the current time. Did I miss something?